### PR TITLE
Feature/tax extension

### PR DESCRIPTION
**Pull request by ERussel**
There is a need to have tax besides of fee logic. The difference is that tax is included into amount the user enters. To provide the logic one need to customise display interface for tax/fee cases and provide opportunity to setup different transaction limits (for example, for 0.01 tax we can send at least 2 tokens in case we have zero precision).

- transaction settings factory add
- fee display settings factory add
- new tax type add to FeeType
- Japanese language support add
- tests improved

### CODE REVIEW

File: CommonWallet/Classes/Common/Input/MoneyPresentable.swift [hunk 1]
```diff
@@ -9,7 +9,7 @@ import Foundation
 protocol MoneyPresentable {
     var formatter: NumberFormatter { get }
     var amount: String { get }
-    var precision: UInt8 { get }
+    var precision: Int16 { get }
```
Comment (unklayman) [inline CommonWallet/Classes/Common/Input/MoneyPresentable.swift:12]: восьми не хватает?

File: CommonWallet/Classes/Common/Service/Network/Middleware/MiddlewareOperationFactoryProtocol.swift [hunk 2]
```diff
@@ -136,20 +136,24 @@ public extension MiddlewareOperationFactoryProtocol {
...
                 .transferAsset(info.source,
                                destinationAccount: info.destination,
                                assetId: info.asset,
                                description: info.details,
-                               amount: info.amount)
+                               amount: amount)
 
             if let fee = info.fee, let feeAccountId = info.feeAccountId {
-                let feeDescription = L10n.Common.transferFee
+                let feeDescription = "transfer fee"
```
Comment (unklayman) [inline CommonWallet/Classes/Common/Service/Network/Middleware/MiddlewareOperationFactoryProtocol.swift:149]: почему хардкод?

**Review by unklayman — APPROVED** [review]

File: CommonWallet/Classes/Common/Input/MoneyPresentable.swift (hunk 1 above)
Comment (ERussel) [reply CommonWallet/Classes/Common/Input/MoneyPresentable.swift:12]: не, просто нативно используется Int16 и чтобы постоянно не кастить решил поменять.

File: CommonWallet/Classes/Common/Service/Network/Middleware/MiddlewareOperationFactoryProtocol.swift (hunk 2 above)
Comment (ERussel) [reply CommonWallet/Classes/Common/Service/Network/Middleware/MiddlewareOperationFactoryProtocol.swift:149]: требования бэкенда, который используется по умолчанию

**Pull request merged**